spyder/plugins/editor/widgets/tests/fixtures.py

```diff
@@ -68,19 +47,42 @@ def setup_editor(qtbot):
     return editorStack, finfo.editor
 
 
-@pytest.fixture
-def lsp_codeeditor(qtbot):
-    """CodeEditor instance with LSP services activated."""
+@pytest.fixture(scope="module")
+def qtbot_module(qapp, request):
+    """Module fixture for qtbot."""
+    result = QtBot(request)
+    return result
+
+
+@pytest.fixture(scope="module")
+def lsp_manager(qtbot_module):
+    """Create an LSP manager instance."""
     # Activate pycodestyle and pydocstyle
     CONF.set('lsp-server', 'pycodestyle', True)
     CONF.set('lsp-server', 'pydocstyle', True)
 
-    # Tell CodeEditor to use introspection
+    # Create the manager
     os.environ['SPY_TEST_USE_INTROSPECTION'] = 'True'
+    manager = LSPManager(parent=None)
 
-    # Create an LSPManager instance to be able to start an LSP client
-    lsp_manager = LSPManager(parent=None)
+    with qtbot_module.waitSignal(manager.sig_initialize, timeout=30000) as blocker:
+        manager.start_client('python')
 
+    settings, language = blocker.args
+    manager.clients[language]['server_settings'] = settings
+
+    yield manager
+
+    # Tear down operations
```

Comment:
I suggest we use the `request.addfinalizer(fin)` syntax as it makes the fixture code more clear and as they state in the docs 

https://docs.pytest.org/en/latest/fixture.html

>Both yield and addfinalizer methods work similarly by calling their code after the test ends, but addfinalizer has two key differences over yield:

>It is possible to register multiple finalizer functions.


>Finalizers will always be called regardless if the fixture setup code raises an exception. This is handy to properly close all resources created by a fixture even if one of them fails to be created/acquired:



Example:
```python
# content of conftest.py
import smtplib
import pytest


@pytest.fixture(scope="module")
def smtp_connection(request):
    smtp_connection = smtplib.SMTP("smtp.gmail.com", 587, timeout=5)

    def fin():
        print("teardown smtp_connection")
        smtp_connection.close()

    request.addfinalizer(fin)
    return smtp_connection  # provide the fixture value
```

Reply:
Thanks for the suggestion. I added it to fixture.